src/benchmark/file_io_write_micro_benchmark.cpp

```diff
@@ -26,64 +22,67 @@ class FileIOWriteMicroBenchmarkFixture : public MicroBenchmarkBasicFixture {
     data_to_write = std::vector<uint32_t>(vector_element_count, VALUE_TO_WRITE);
     control_sum = vector_element_count * uint64_t{VALUE_TO_WRITE};
 
-    if (creat("file.txt", O_WRONLY) < 1) {
-      std::cout << "create error" << std::endl;
-    }
-    chmod("file.txt", S_IRWXU);  // enables owner to rwx file
-  }
-
-  void sanity_check(uint32_t NUMBER_OF_BYTES ) {
-    auto fd = int32_t{};
-    if ((fd = open("file.txt", O_RDONLY)) < 0) {
-      std::cout << "open error " << std::strerror(errno) << std::endl;
+    if (creat(filename, O_WRONLY) < 1) {
+	    Fail("Create error:" + std::strerror(errno));
     }
-    chmod("file.txt", S_IRWXU);  // enables owner to rwx file
-    std::vector<uint32_t> read_data;
-    read_data.resize(NUMBER_OF_BYTES / sizeof(uint32_t));
-
-    // Getting the mapping to memory.
-    off_t OFFSET = 0;
-
-    uint32_t* map = reinterpret_cast<uint32_t*>(mmap(NULL, NUMBER_OF_BYTES, PROT_READ, MAP_PRIVATE, fd, OFFSET));
-    if (map == MAP_FAILED) {
-      std::cout << "Mapping for Sanity Check Failed." << std::strerror(errno) << std::endl;
-    }
-
-    memcpy(std::data(read_data), map, NUMBER_OF_BYTES);
-    const auto sum = std::accumulate(read_data.begin(), read_data.end(), uint64_t{0});
-    Assert(control_sum == sum, "Sanity check failed. Got: " + std::to_string(sum) + "Expected: " + std::to_string(control_sum));
-    // Remove memory mapping after job is done.
-    if (munmap(map, NUMBER_OF_BYTES) != 0) {
-      std::cout << "Unmapping for Sanity Check failed." << std::endl;
-    }
-
-    close(fd);
+    chmod(filename, S_IRUSR | S_IWUSR);  // enables owner to read and write file
   }
 
   void TearDown(::benchmark::State& /*state*/) override {
-    // TODO(phoeinx): Error handling
-    std::remove("file.txt");
+    Assert(std::remove(filename) == 0, "Remove error: " + std::strerror(errno));
   }
 
  protected:
   std::vector<uint32_t> data_to_write;
-  void mmap_write_benchmark(benchmark::State& state, const int flag, int data_access_mode, const int32_t file_size);
+	uint64_t control_sum = uint64_t{0};
+	uint32_t vector_element_count;
+	uint32_t VALUE_TO_WRITE = 42;
+	const char* filename = "file.txt"; //const char* needed for C-System Calls
+
+  void mmap_write_benchmark(benchmark::State& state, const int flag, int data_access_mode, const ssize_t file_size);
+  void sanity_check(uint32_t NUMBER_OF_BYTES);
 };
 
-BENCHMARK_DEFINE_F(FileIOWriteMicroBenchmarkFixture, WRITE_NON_ATOMIC)(benchmark::State& state) {  // open file
+
+void FileIOWriteMicroBenchmarkFixture::sanity_check(uint32_t NUMBER_OF_BYTES) {
+	auto fd = int32_t{};
+	if ((fd = open(filename, O_RDONLY)) < 0) {
+		close(fd);
+		Fail("Open error:" + std::strerror(errno));
+	}
+
+	auto read_data = std::vector<uint32_t>(NUMBER_OF_BYTES / sizeof(uint32_t));
+
+	const off_t OFFSET = 0;
+	uint32_t* map = reinterpret_cast<uint32_t*>(mmap(NULL, NUMBER_OF_BYTES, PROT_READ, MAP_PRIVATE, fd, OFFSET));
```

Comment:
we can also use auto on pointers :)
```suggestion
	auto* map = reinterpret_cast<uint32_t*>(mmap(NULL, NUMBER_OF_BYTES, PROT_READ, MAP_PRIVATE, fd, OFFSET));
```